crates/bdk/src/wallet/mod.rs

```diff
@@ -2312,6 +2312,55 @@ impl<D> Wallet<D> {
     pub fn local_chain(&self) -> &LocalChain {
         &self.chain
     }
+
+    /// Set lookahead for all keychains.
+    ///
+    /// Passing a lookahead of 0 will result in an error. This is because if we sync
+    /// the chain with a lookahead of 0, we would not find any update since we don't
+    /// have any scripts stored.
+    pub fn set_lookahead_for_all(&mut self, lookahead: u32) -> Result<(), Error> {
+        if lookahead == 0 {
+            return Err(Error::Generic(
+                "lookahead must be greater than 0".to_string(),
+            ));
+        }
+        self.indexed_graph.index.set_lookahead_for_all(lookahead);
+        Ok(())
+    }
+
+    /// Insert all the block's relevant transactions into the IndexedTxGraph.
+    pub fn apply_block_relevant(
```

Comment:
My main concern with this PR is we don't have a plan for the suite of methods we'll need as I talked about over in https://github.com/vladimirfomene/bdk/pull/2#issuecomment-1823605394

If we don't implement them in this PR we should at least have a plan so we don't have to rename things later and do proper documentation for this method.

- `apply_block(block, height)` applies the block when the previous block is in the wallet otherwise errors.
- `apply_block_connected_to(block, height, connected_to)` applies the block assuming that the previous block hash of the block is in a descendant of `connected_to`. Will error out of `connected_to` is not in the wallet (it's fine for `connected_to` to be the previous block hash -- then this is the same as `apply_block`).
- `apply_block_assume_connected(block, height)` -- first calls `apply_block` internally but if that was not connected then assumes that the block does connect to the existing chain and so finds the right spot.

Does this make sense? If so what we have to do here is improve the documentation for this method and add these other methods in another PR.

Another question is how these fit in with the "batch insert" methods. Can we expose those to wallet users in addition to (or instead of) the above?

Reply:
@LLFourn after reading your points in #1245, I agree with this. However, can we not have `apply_block_assume_connected`? I think the emitter should always be explicit with which block to connect to.